backend/dataall/modules/maintenance/services/maintenance_service.py

```diff
@@ -0,0 +1,153 @@
+        try:
+            with engine.scoped_session() as session:
+                maintenance_record = MaintenanceRepository(session).get_maintenance_record()
+                if (
+                    maintenance_record.status == MaintenanceStatus.PENDING.value
+                    or maintenance_record.status == MaintenanceStatus.ACTIVE.value
+                ):
+                    logger.error(
+                        'Maintenance window already in PENDING or ACTIVE state. Cannot start maintenance window. Stop the maintenance window and start again'
+                    )
+                    return False
+                MaintenanceRepository(session).save_maintenance_status_and_mode(
+                    maintenance_status=MaintenanceStatus.PENDING.value, maintenance_mode=mode
+                )
+            # Disable scheduled ECS tasks
+            # Get all the SSM Params related to the scheduled tasks
+            ecs_scheduled_rules_list = MaintenanceService._get_ecs_rules()
+            event_bridge_session = EventBridge(region=os.getenv('AWS_REGION', 'eu-west-1'))
+            event_bridge_session.disable_scheduled_ecs_tasks(ecs_scheduled_rules_list)
+            return True
+        except Exception as e:
+            logger.error(f'Error occurred while starting maintenance window due to {e}')
+            return False
+
+    @staticmethod
+    def stop_maintenance_window(engine, groups=None):
+        """
+        Stop maintenance window by performing following actions
+            1. Perform validation to check if the user belongs to the DAAdministrators group
+            2. Update the RDS table by changing the status to INACTIVE and mode to '-'
+            3. Enable all data.all related ECS scheduled tasks
+        @param engine: db engine
+        @param groups: user groups from context.groups
+        @return: return True if successful or False
+        """
+
+        # Check from the context if the groups contains the DAAAdminstrators group
+        if groups is None:
+            groups = []
+        if not TenantPolicyValidationService.is_tenant_admin(groups):
+            raise Exception('Only data.all admin group members can stop maintenance window')
+        logger.info('Stopping maintenance mode')
+        try:
+            with engine.scoped_session() as session:
+                maintenance_record = MaintenanceRepository(session).get_maintenance_record()
+                if maintenance_record.status == MaintenanceStatus.INACTIVE.value:
+                    logger.error('Maintenance window already in INACTIVE state. Cannot stop maintenance window')
+                    return False
+                MaintenanceRepository(session).save_maintenance_status_and_mode(
+                    maintenance_status='INACTIVE', maintenance_mode=''
+                )
+            # Enable scheduled ECS tasks
+            ecs_scheduled_rules_list = MaintenanceService._get_ecs_rules()
```

Comment:
nit: rename event_bridge_session to event_bridge_client. I have missunderstood this line twice hahahahh because it says session I understood it returns a boto3 session and then the session is used to directly call the boto3 function

Reply:
We have both types, I don't mind event_bridge is fine as well